commerce_coordinator/apps/frontend_app_ecommerce/filters.py

```diff
@@ -0,0 +1,29 @@
+"""
+Filters used by the frontend_app_ecommerce app
+"""
+from openedx_filters.tooling import OpenEdxPublicFilter
+
+
+class OrderHistoryRequested(OpenEdxPublicFilter):
+    """
+    Filter to gather order data from the defined PipelineStep(s)
+    """
+    # See pipeline step configuration OPEN_EDX_FILTERS_CONFIG dict in `settings/base.py`
+    filter_type = "org.edx.coordinator.frontend_app_ecommerce.order.history.requested.v1"
+
+    @classmethod
+    def run_filter(cls, params, order_data=None):
+        """
+        Call the PipelineStep(s) defined for this filter, to gather orders and return together
+        Arguments:
+            params: arguments passed through from the original order history url querystring
+            order_data: any preliminary orders we want to append to when running the pipeline
+        """
+        if order_data is None:
+            order_data = []
+
+        pipeline_data = super().run_pipeline(params=params, order_data=order_data)
+        result = pipeline_data.get("order_data")
+
+        # Note: OpenEdxPublicFilter run_pipeline returns an array with our dictionary inside
+        return result[0]
```

Comment:
Since the OpenEdxPublicFilter run_pipeline (super) returns an array with our dictionary inside rather than the dictionary itself, does anyone know a nicer-looking idiom for returning the contents of the array, or is this the best we can do?

Reply:
I think it's great as-is. Might be a small chance of a future dev footgun here once `result` has more than one item (i.e. after adding a new order history source), but an easy `assert len(result) == 1` will remind us to refactor this later.

Other ideas if we really do want to avoid this:
* Make `order_data` a dictionary, with keys being the service from which the `order_data` originates. Then, this line would be `return result['ecommerce']`.
* Add a pipeline step that merges a `list(list(dict()))` into a `list(dict())`. Would require a `json.loads()` to transform `order_data` from a JSON string to a Python list of dictionaries.